### PR TITLE
[CP-669] Duplicate in network name

### PR DESCRIPTION
**Pull request by wercix**
Jira: [CP-669]

**Description**
During the tests we fined out another duplicate - T-mobile
<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-669]: https://appnroll.atlassian.net/browse/CP-669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by igorbezsmertnyi** [comment]
@wercix what do you think about this kind of solution https://codezup.com/find-duplicate-words-in-a-string-interview-challenge/ ?

**Comment by dkarski** [comment]
> @wercix what do you think about this kind of solution https://codezup.com/find-duplicate-words-in-a-string-interview-challenge/ ?

I think that can occur the name with having a duplicate in their names? 😅 

However this is less probable to occur, I think that it's a nice idea to implement 👌

**Review by dkarski — COMMENTED** [review]
> @wercix what do you think about this kind of solution https://codezup.com/find-duplicate-words-in-a-string-interview-challenge/ ?

I think that can occur the name with having a duplicate in their names? 😅 

However this is less probable to occur, I think that it's a nice idea to implement 👌

**Comment by wercix** [comment]
Ok, but how about this case with _T-mobile_ and _T-mobile.pl_ ? Find duplicate function will not consider it as a duplicate. So maybe we could leave my solution and add this unction in case of any other duplicates

File: packages/app/src/overview/components/status/status.component.tsx [hunk 1]
```diff
@@ -64,7 +64,16 @@ const Status: FunctionComponent<StatusProps> = ({
   networkLevel = 0,
 }) => {
   const strength = Math.round(networkLevel * 100)
-  const networkNameDisplay = network === "Orange Orange" ? "Orange" : network
+  const displayNetworkName = (name: string) => {
```
Comment (dkarski) [inline packages/app/src/overview/components/status/status.component.tsx:67]:
```suggestion
  const displayNetworkName = (name: string): string => {
```

File: packages/app/src/overview/components/status/status.component.tsx [hunk 2]
```diff
@@ -64,7 +64,16 @@ const Status: FunctionComponent<StatusProps> = ({
...
+    const orange = "Orange Orange"
+    const tmobile = "T-Mobile T-Mobile.pl"
+    const names: { [key: string]: string } = {
```
Comment (dkarski) [inline packages/app/src/overview/components/status/status.component.tsx:70]:
```suggestion
    const names: Record<string, string> = {
```

**Review by dkarski — DISMISSED** [review]

File: packages/app/src/overview/components/status/status.component.tsx [hunk 3]
```diff
@@ -64,7 +64,38 @@ const Status: FunctionComponent<StatusProps> = ({
   networkLevel = 0,
 }) => {
   const strength = Math.round(networkLevel * 100)
-  const networkNameDisplay = network === "Orange Orange" ? "Orange" : network
+  const findDuplicatesInNetworkName = (name: string) => {
```
Comment (dkarski) [inline packages/app/src/overview/components/status/status.component.tsx:67]:
```suggestion
  const findDuplicatesInNetworkName = (name: string): string => {
```

File: packages/app/src/overview/components/status/status.component.tsx [hunk 4]
```diff
@@ -64,7 +64,38 @@ const Status: FunctionComponent<StatusProps> = ({
...
+    const result: string[] = []
+    const wordsArray = name.split(" ")
+    const count: { [key: string]: number } = {}
```
Comment (dkarski) [inline packages/app/src/overview/components/status/status.component.tsx:70]:
```suggestion
    const count:  Record<string, number>= {}
```

**Review by dkarski — APPROVED** [review]

**Review by dkarski — APPROVED** [review]
please add test to cover

**Review by igorbezsmertnyi — APPROVED** [review]

**Pull request merged**